src/index/base.cpp

```diff
@@ -177,17 +177,19 @@ void BaseIndex::ThreadSync()
                 // No need to handle errors in Commit. See rationale above.
                 Commit();
             }
-
-            CBlock block;
-            if (!ReadBlockFromDisk(block, pindex, consensus_params)) {
-                FatalError("%s: Failed to read block %s from disk",
-                           __func__, pindex->GetBlockHash().ToString());
-                return;
-            }
-            if (!WriteBlock(block, pindex)) {
-                FatalError("%s: Failed to write block %s to index database",
-                           __func__, pindex->GetBlockHash().ToString());
-                return;
+            {
+                LOCK(cs_main);
```

Comment:
This is one of two places where ReadBlockFromDisk/WriteBlock were not protected by cs_main.  PowCache is expecting to be protected externally, so we need this lock.